osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -235,11 +235,18 @@ private Task<IBeatmap> loadBeatmapAsync()
                     // Todo: Handle cancellation during beatmap parsing
                     var b = GetBeatmap() ?? new Beatmap();
 
-                    // The original beatmap version needs to be preserved as the database doesn't contain it
-                    BeatmapInfo.BeatmapVersion = b.BeatmapInfo.BeatmapVersion;
-
-                    // Use the database-backed info for more up-to-date values (beatmap id, ranked status, etc)
-                    b.BeatmapInfo = BeatmapInfo;
+                    // Copy across values of key properties for which the database-backed model has data that the decoded beatmap isn't going to.
+                    b.BeatmapInfo.ID = BeatmapInfo.ID;
```

Comment:
The only thing I'm concerned with here is that previously we were doing a reference copy of `BeatmapInfo`, whereas now we're mutating the existing one. Probably fine (better?) but still leaves me a touch unsure.

Reply:
I don't know what your expectation is from me with respect to this comment. I could do five other things (like copy the other way around, for instance), but I dunno what the specific problem here is. If it's being worried about something in gameplay innards is accidentally doing some reference equality check that this is gonna fail, (a) I'm 99% not gonna be able to find it, (b) we're completely cooked with this project and probably need to start over again.

Reply:
Reference checks or mutation in some weird place.

But yeah, no action expected. We all know how fiddly the code is around `IBeatmap` / `BeatmapInfo` etc